gandalf.js

```diff
@@ -1,118 +1,115 @@
 /**
```

Comment:
SCRIPT-Well done with the efficient error handling, it covers all the possible scenarios.